client/src/main.rs

```diff
@@ -1,33 +1,42 @@
 use std::error::Error;
 
-use clap::{App, Arg, SubCommand};
+use clap::{App, AppSettings, Arg, SubCommand};
+use failure::Fail;
+use indradb::{
+    EdgeKey, EdgePropertyQuery, EdgeQuery, SpecificEdgeQuery, SpecificVertexQuery, VertexPropertyQuery, VertexQuery,
+};
+use indradb_proto as proto;
+use std::convert::TryInto;
 
 #[tokio::main]
 pub async fn main() -> Result<(), Box<dyn Error>> {
-    let vertex_query_arg = Arg::with_name("query")
-        .help("the JSON vertex query")
-        .required(true)
-        .index(1);
+    let vertex_id_arg = Arg::with_name("uuid")
+        .help("the UUID of the target vertex")
+        .required(true);
+    let outbound_id_arg = Arg::with_name("outbound_id")
+        .help("the outbound vertex ID")
+        .required(true);
+    let edge_type_arg = Arg::with_name("type").help("the edge type").required(true);
+    let inbound_id_arg = Arg::with_name("inbound_id")
+        .help("the inbound vertex ID")
+        .required(true);
```

Comment:
Do these args require either a `.takes_value(true)` or `.index(true)`? Maybe they don't, not sure.